### PR TITLE
[ANOMALY CLASSIFICATION] Improve early stopping for STFPM

### PR DESCRIPTION
**Pull request by djdameln**
# Description
This PR improves early stopping for the STFPM Anomaly Classification algorithm.

# Changes:
- Make early stopping patience configurable
- Increase default early stopping patience from 3 epochs to 10 epochs
- Change default early stopping metric to F1 score
- Re-generate configuration yaml files

# Build
- [VM]( https://ci.iotg.sclab.intel.com/job/IMPT/job/IMPTOps/23330/) ✅
- [BMM](https://ci.iotg.sclab.intel.com/job/IMPT/job/IMPTOps/23351/) ✅
- Manual Training ✅

### CODE REVIEW

File: external/anomaly/anomaly_classification/configs/stfpm/configuration.yaml [hunk 1]
```diff
@@ -113,7 +124,6 @@ pot_parameters:
       operator: AND
       rules: []
       type: UI_RULES
-    value: 300
```
Comment (ashwinvaidya17) [inline external/anomaly/anomaly_classification/configs/stfpm/configuration.yaml:116]: Why is this removed?

**Review by ashwinvaidya17 — DISMISSED** [review]
Thanks for the efforts. I only have one question.

File: external/anomaly/anomaly_classification/configs/stfpm/configuration.yaml (hunk 1 above)
Comment (djdameln) [reply external/anomaly/anomaly_classification/configs/stfpm/configuration.yaml:116]: I used [this script](https://gitlab.devtools.intel.com/vmc-eip/IMPT/impt/-/blob/develop/src/sc_sdk/ote_hyperparam_configuration_generator) from SC_SDK to generate the `configuration.yaml` files. Apparently some changes were made in how the yaml files get populated in this conversion process. I don't think it should be a problem, because the `default_value` field is still present.

**Comment by djdameln** [comment]
run OTE_SDK tests

File: external/anomaly/anomaly_classification/configs/stfpm/configuration.yaml (hunk 1 above)
Comment (ashwinvaidya17) [reply external/anomaly/anomaly_classification/configs/stfpm/configuration.yaml:116]: Alright thanks!

**Comment by samet-akcay** [comment]
run ote_sdk tests

**Review by samet-akcay — APPROVED** [review]

**Review by ashwinvaidya17 — APPROVED** [review]

**Comment by samet-akcay** [comment]
@Ilya-Krylov is it possible to merge this PR? The target branch has somehow been changed to develop. We would like to merge this to the release. They are synced, right?

**Comment by Ilya-Krylov** [comment]
@samet-akcay we do not merge to releases/v0.1.1 branch anymore

**Review by druzhkov-paul — APPROVED** [review]
